src/main/java/com/aws/greengrass/shadowmanager/sync/RequestBlockingQueue.java

```diff
@@ -369,9 +369,10 @@ public SyncRequest poll(long timeout, TimeUnit unit) throws InterruptedException
      * Take an item from the queue. Returns immediately if no item is available.
      *
      * @return the request at the head of the queue or null if no items are available.
+     * @throws InterruptedException if the thread is interrupted while waiting
      */
-    public SyncRequest poll() {
-        lock.lock();
+    public SyncRequest poll() throws InterruptedException {
```

Comment:
why make this change? It was intentionally copying the java BlockingQueue interface  (but not implementing since it's not a collection) where `poll` is not interruptible unless you specify a timeout.

It returns immediately if there's no data, so there's no need to wait or interrupt.